core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel_MDRange.hpp

```diff
@@ -0,0 +1,457 @@
+      typename Policy::point_type offset, const FunctorType& functor,
+      const Policy& policy) const {
+    const ptrdiff_t begin_0 = offset[0];
+    ptrdiff_t end_0         = begin_0 + policy.m_tile[0];
+    end_0 = end_0 < policy.m_upper[0] ? end_0 : policy.m_upper[0];
+
+    const ptrdiff_t begin_1 = offset[1];
+    ptrdiff_t end_1         = begin_1 + policy.m_tile[1];
+    end_1 = end_1 < policy.m_upper[1] ? end_1 : policy.m_upper[1];
+
+    const ptrdiff_t begin_2 = offset[2];
+    ptrdiff_t end_2         = begin_2 + policy.m_tile[2];
+    end_2 = end_2 < policy.m_upper[2] ? end_2 : policy.m_upper[2];
+
+    const ptrdiff_t begin_3 = offset[3];
+    ptrdiff_t end_3         = begin_3 + policy.m_tile[3];
+    end_3 = end_3 < policy.m_upper[3] ? end_3 : policy.m_upper[3];
+
+    const ptrdiff_t begin_4 = offset[4];
+    ptrdiff_t end_4         = begin_4 + policy.m_tile[4];
+    end_4 = end_4 < policy.m_upper[4] ? end_4 : policy.m_upper[4];
+
+#pragma omp for collapse(5)
+    for (ptrdiff_t i0 = begin_0; i0 < end_0; i0++)
+      for (ptrdiff_t i1 = begin_1; i1 < end_1; i1++)
+        for (ptrdiff_t i2 = begin_2; i2 < end_2; i2++)
+          for (ptrdiff_t i3 = begin_3; i3 < end_3; i3++)
+            for (ptrdiff_t i4 = begin_4; i4 < end_4; i4++)
+              functor(i0, i1, i2, i3, i4);
+  }
+
+  template <int Rank>
+  inline typename std::enable_if<Rank == 6>::type execute_tile(
+      typename Policy::point_type offset, const FunctorType& functor,
+      const Policy& policy) const {
+    const ptrdiff_t begin_0 = offset[0];
+    ptrdiff_t end_0         = begin_0 + policy.m_tile[0];
+    end_0 = end_0 < policy.m_upper[0] ? end_0 : policy.m_upper[0];
+
+    const ptrdiff_t begin_1 = offset[1];
+    ptrdiff_t end_1         = begin_1 + policy.m_tile[1];
+    end_1 = end_1 < policy.m_upper[1] ? end_1 : policy.m_upper[1];
+
+    const ptrdiff_t begin_2 = offset[2];
+    ptrdiff_t end_2         = begin_2 + policy.m_tile[2];
+    end_2 = end_2 < policy.m_upper[2] ? end_2 : policy.m_upper[2];
+
+    const ptrdiff_t begin_3 = offset[3];
+    ptrdiff_t end_3         = begin_3 + policy.m_tile[3];
+    end_3 = end_3 < policy.m_upper[3] ? end_3 : policy.m_upper[3];
+
+    const ptrdiff_t begin_4 = offset[4];
+    ptrdiff_t end_4         = begin_4 + policy.m_tile[4];
+    end_4 = end_4 < policy.m_upper[4] ? end_4 : policy.m_upper[4];
+
+    const ptrdiff_t begin_5 = offset[5];
+    ptrdiff_t end_5         = begin_5 + policy.m_tile[5];
+    end_5 = end_5 < policy.m_upper[5] ? end_5 : policy.m_upper[5];
+
+#pragma omp for collapse(6)
+    for (ptrdiff_t i0 = begin_0; i0 < end_0; i0++)
+      for (ptrdiff_t i1 = begin_1; i1 < end_1; i1++)
+        for (ptrdiff_t i2 = begin_2; i2 < end_2; i2++)
+          for (ptrdiff_t i3 = begin_3; i3 < end_3; i3++)
+            for (ptrdiff_t i4 = begin_4; i4 < end_4; i4++)
+              for (ptrdiff_t i5 = begin_5; i5 < end_5; i5++)
+                functor(i0, i1, i2, i3, i4, i5);
+  }
```

Comment:
Isn't the maximum rank 8?

Reply:
Not everything implemented yet ...

Reply:
actually the implementation of MDRangePolicy here is pretty bad. It is just what I need to get the View initialization going for most of the normal testing of RangePolicy, etc.

Reply:
Adding 7 and 8